### PR TITLE
FEATURE: Show a confirmation modal when "Dismiss all" would dismiss an important notification

### PR DESCRIPTION
**Pull request by nattsw**
showing the dismiss all notifications confirmation modal when there're important notifications

### CODE REVIEW

File: app/assets/javascripts/discourse/app/templates/modal/dismiss-notification-confirmation.hbs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{{#d-modal-body headerClass="hidden" class="dismiss-notification-confirmation"}}
+    {{i18n "notifications.dismiss_confirmation.body" count=count}}
+{{/d-modal-body}}
+
+<div class="modal-footer">
+  {{d-button icon="check" class="btn-primary" action=(action "confirm") label="notifications.dismiss_confirmation.confirm"}}
+  {{d-button action=(route-action "closeModal") label="notifications.dismiss_confirmation.cancel" class="cancel"}}
+</div>
```
Comment (nattsw) [inline app/assets/javascripts/discourse/app/templates/modal/dismiss-notification-confirmation.hbs:8]: Some useful input would be if the modals have a standard with regards to titles, bodies, and buttons. The style guide doesn't seem to have an opinion.

Comment (martin-brennan) [reply app/assets/javascripts/discourse/app/templates/modal/dismiss-notification-confirmation.hbs:8]: Oh sorry but you can use a `bootbox.confirm` for this, see:

https://github.com/discourse/discourse/blob/08e625c4469f297de21cbdfe78d133e0d550fd2a/app/assets/javascripts/admin/addon/controllers/admin-customize-themes-show.js#L392-L401

Comment (martin-brennan) [reply app/assets/javascripts/discourse/app/templates/modal/dismiss-notification-confirmation.hbs:8]: Then you can delete the other modal stuff you have added.

File: app/assets/javascripts/discourse/app/widgets/user-menu.js [hunk 2]
```diff
@@ -249,7 +250,18 @@ export default createWidget("user-menu", {
   },
 
   dismissNotifications() {
-    return this.state.markRead();
+    const unreadHighPriorityNotifications = this.currentUser.get(
+      "unread_high_priority_notifications"
+    );
+
+    if (unreadHighPriorityNotifications > 0) {
+      return showModal("dismiss-notification-confirmation").setProperties({
```
Comment (martin-brennan) [inline app/assets/javascripts/discourse/app/widgets/user-menu.js:258]: Here is where you can do bootbox.confirm

**Review by martin-brennan — CHANGES_REQUESTED** [review]
Looks great, you just need to change to bootbox :)

**Comment by martin-brennan** [comment]
@nattsw You have a linting issue (which is in the template you are removing), but these lints should be run in a pre-commit hook, did they not run for you? See .git/hooks/pre-commit

**Comment by nattsw** [comment]
We probably need some docs for our git hooks. 🤔

File: app/assets/javascripts/discourse/tests/acceptance/dismiss-notification-modal-test.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    await visit("/");
+    await click(".current-user");
+    await click(".notifications-dismiss");
+    assert.ok(exists(".bootbox.modal"));
+
+    assert.strictEqual(
+      query(".bootbox.modal .modal-body").innerText,
+      "You have 2 important notifications, are you sure you would like to dismiss?"
+    );
+    await click(".bootbox.modal .btn-default");
```
Comment (nattsw) [inline app/assets/javascripts/discourse/tests/acceptance/dismiss-notification-modal-test.js:38]: Just a weird test finding when using the bootbox vs a modal:

When creating 2 tests in sequence, using a regular modal doesn't result in test contamination if opening the modals in both tests

But for the same 2 tests, using bootbox results in test contamination for the second test if the first test's bootbox wasn't closed. So we have to be mindful and close it after we're done with it here.

**Comment by nattsw** [comment]
confirm dismiss notifications modal using a bootbox

It seems the bootbox renders weirdly for iPhone X? I'll probably spend more time looking at the other occurrences and see if it's the same.

File: app/assets/javascripts/discourse/app/widgets/user-menu.js [hunk 4]
```diff
@@ -249,7 +251,26 @@ export default createWidget("user-menu", {
   },
 
   dismissNotifications() {
-    return this.state.markRead();
+    const unreadHighPriorityNotifications = this.currentUser.get(
```
Comment (martin-brennan) [inline app/assets/javascripts/discourse/app/widgets/user-menu.js:254]: This is a nitpick more than anything, sorry I didn't notice yesterday, but we don't need to use ember's `.get` for single properties like this, only chained properties. e.g. `currentUser.get("options.some_other_thing")`

**Review by martin-brennan — APPROVED** [review]
LGTM :)

**Pull request merged**